src/utils/fuzzySearch.js

```diff
@@ -0,0 +1,38 @@
+import Fuse from "fuse.js";
```

Comment:
We had previously used `fuse.js` and reverted it to reduce the total bundle size. Can you try implementing the similar fuzzy logic without involving an external package?